src/interfaces/widgets/layout.py

```diff
@@ -0,0 +1,49 @@
+"""
+
+    This module generate a Qt layout wrapper object to arrange child widgets
+    within a widget to ensure that they make good use of the available space.
+
+"""
+
+from PyQt5.QtWidgets import QLayout, QHBoxLayout, QVBoxLayout, QStackedLayout
+
+
+class Layout(QLayout):
+
+    def __init__(self, widget, **kwargs):
+        super(Layout, self).__init__(widget)
+
+        # Lambda func grab input args
+        get_param = lambda x : kwargs.get(x)
+
+        l_m = get_param("l_m")
```

Comment:
These margin values should probably have default values as the call to `self.setContentsMargins(...)` below fails if they're `None`.

Reply:
Moreover, I think it's a better idea to instead of conditional set values based on whether they're given, set sensible defaults that can be overridden.

So instead of adding checks like this:
```python
l_m and t_m and r_m and b_m and self.setContentsMargins(l_m, t_m, r_m, b_m)
```

Define defaults like this:
```python
        l_m = get_param("l_m") or 0
        t_m = get_param("t_m") or 0
        r_m = get_param("r_m") or 0
        b_m = get_param("b_m") or 0
        space = get_param("space") or 0
```

Or better yet, define all the expected parameters in the `__init__` call so someones editor knows what to expect when defining an instance.

Reply:
fixed

Reply:
Cool, great suggestion. Updated!